protocol.md

```diff
@@ -0,0 +1,249 @@
+`timestamp` is Unix time calculated when the message is created (TODO: does it come from Whisper or the system?)
+
+`clock` is calculated using the algorithm of [Lamport timestamps](https://en.wikipedia.org/wiki/Lamport_timestamps). When there are messages available in a chat, `clock`'s value is calculated as `last-message-clock-value + 1`. If there are no messages, `clock` is initialized with `timestamp`'s value.
+
+`clock` value is used for the message ordering. Due to the used algorithm, distributed nature of the system and physics, we achieve casual ordering which might produce counterintuitive results in some edge cases. For example, when one joins a public chat and sends a message before receiving the exist messages, their message `clock` value might be lower and the message will end up in the past when the historical messages are fetched.
+
+## Quoting
+
+TBD
+
+# Whisper adapter
+
+Whisper has been chosen as an messages exchange protocol because it was designed as an off-chain communcation layer for the Ethereum nodes. It supports e2e encryption and uses epidemic spread to route data to all members of the network. It also provides [darkness to some extent](https://github.com/ethereum/go-ethereum/wiki/Achieving-Darkness).
+
+However, Whisper was not designed to handle huge number of messages and real-time communication. These are the tradeoffs that we accepted when implementing this protocol.
+
+This protocol can either work using a Whisper service which requires a protocol implementation to run in the same process as a Whisper node as well as using a Whisper client which might run as a separate Ethereum node and communicate through IPC or WebSocket.
+
+There is some tight coupling between the payload and Whisper:
+* Whisper message topic depends on the actual message type (see [Topic](#topic))
+* Whisper message uses a different key (asymmetric or symmetric) depending on the actual message type (see [Keys management](#keys-management))
+
+## Whisper node configuration
+
+If you want to run a Whisper node and receive messages from Status clients, you need to configure it properly, otherwise, some message might not be received.
+
+Whisper's Proof Of Work algorithm is used to to deter denial of service and various spam/flood attacks against the Whisper network. The sender of a message must perform some work which in this case means performing some computations. Because Status main client is a mobile client, this easily leads to battery draining and poor performance of the app itself. Hence, these clients send envelopes with the following settings:
+* proof-of-work equal `0.002`
+* proof-of-work time equal to `10`
+* time-to-live equal `10`
+
+If you want to receive messages from a mobile client, you need to run Whisper node with PoW set to `0.002`. In case of `geth`, this option can be overriden with `-shh.pow=0.002` flag.
+
+TODO: provide an instruction how to start a Whisper node with proper configuration using geth.
+
+## Keys management
```

Comment:
@corpetty could you help out here wrt key paths etc? Thinking re keycard too

Reply:
yeah, Keycard def has the most up to date info on what the changes will be when keycard is integrated (separate keys).  I'll follow up with them and make sure the schematics I have for it are up to date.

Reply:
But we should describe the current flow here. I am not sure when the keycard things is gonna happen, also, does it gonna change anything for the current users who do not have keycard?

Reply:
FYI updated this section.